endpoints/openrtb2/auction.go

```diff
@@ -1265,21 +1289,33 @@ func writeError(errs []error, w http.ResponseWriter, labels *pbsmetrics.Labels)
 	return rc
 }
 
-// Returns the effective publisher ID
-func effectivePubID(pub *openrtb.Publisher) string {
-	if pub != nil {
-		if pub.Ext != nil {
-			var pubExt openrtb_ext.ExtPublisher
-			err := json.Unmarshal(pub.Ext, &pubExt)
-			if err == nil && pubExt.Prebid != nil && pubExt.Prebid.ParentAccount != nil && *pubExt.Prebid.ParentAccount != "" {
-				return *pubExt.Prebid.ParentAccount
-			}
+// Sets the effective publisher ID
+func setEffectivePubID(req *openrtb.BidRequest, isAmp bool, urlValues url.Values) {
+	var pub *openrtb.Publisher
+	if req.App != nil {
+		if req.App.Publisher == nil {
+			req.App.Publisher = new(openrtb.Publisher)
 		}
-		if pub.ID != "" {
-			return pub.ID
+		pub = req.App.Publisher
+	}
+	if req.Site != nil {
+		if req.Site.Publisher == nil {
+			req.Site.Publisher = new(openrtb.Publisher)
 		}
+		pub = req.Site.Publisher
+	}
+
+	if pub.ID == "" && isAmp {
+		// For amp requests, the publisher ID could be sent via the account
+		// query string
+		if acc := urlValues.Get("account"); acc != "" && acc != "ACCOUNT_ID" {
+			pub.ID = acc
+		}
+	}
+
+	if pub.ID == "" {
+		pub.ID = pbsmetrics.PublisherUnknown
```

Comment:
This will cause us to add our unknown publisher value in metrics as the publisher in the request, rather than leaving it unset. This will cause confusion in bidders who do not know that this particular string is PBS metrics slang for "unknown publisher" rather than a publisher explicitly called out with an unusual ID, who may end up appearing to have a lot of diverse inventory. I think we want unknown publishers to remain with an unset ID in the request, as that is the normal signal for an unknown publisher.